dags/iop/parser.py

```diff
@@ -100,3 +105,80 @@ def _get_extracted_arxiv_eprint_value(self, article):
             self.logger.error("The arXiv value is not valid.", dois=self.dois)
         except AttributeError:
             self.logger.error("No arXiv eprints found", dois=self.dois)
+
+    def _get_authors(self, article):
+        contrib_types = article.findall(
+            "front/article-meta/contrib-group/contrib[@contrib-type='author']"
+        )
+        authors = []
+        surname = ""
+        given_names = ""
+
+        for contrib_type in contrib_types:
+            try:
+                surname = contrib_type.find("name/surname").text
+            except AttributeError:
+                self.logger.error("Surname is not found in XML", dois=self.dois)
+            try:
+                given_names = contrib_type.find("name/given-names").text
+            except AttributeError:
+                self.logger.error("Given_names is not found in XML", dois=self.dois)
+
+            reffered_ids = contrib_type.findall("xref[@ref-type='aff']")
+            affiliations = [
+                self._get_affiliation_value(article, reffered_id)
+                for reffered_id in reffered_ids
+                if self._get_affiliation_value(article, reffered_id) is not None
+            ]
+
+            if "collaboration" not in given_names.lower():
+                author = clean_dict(
+                    {
+                        "surname": surname,
+                        "given_names": given_names,
+                        "affiliations": affiliations,
+                    }
+                )
+                authors.append(author)
+
+        if all(authors):
```

Comment:
I think that you are doing an unnecessary step here, I would check before if the required data (name, given_name, and affiliations) was retrieved and if not, I would raise an error. Also, I think there's no difference between returning `None` (as you do now) and empty list (default value for authors)

Reply:
the thing that neither the name, given_name, and neither affiliations are required. Just the field 'authors' is required. In the end, the outputs such as:
```
authors = [{"name": "Name"}]
authors = [{"name": "Name", "given_name": "Given name"}]
authors = [{"name": "Name", "given_name": "Given name", "affiliations": [{"country": "Blabla", "institution": "Bla"}]}]
authors = [{"name": "Name", "given_name": "Given name", "affiliations": [{"country": "Blabla"}]}]
authors = [{"affiliations": [{"country": "Blabla", "institution": "Bla"}]}]
```

Reply:
Ok, but I would handle it a different way than rewriting dict. You can add entries to the dict conditionally etc., there are many options ;).